File: 2018_06__Julia_my_new_optimization_friend__introduction_for_MATLAB_users__at_IETR_seminar/examples/Solving_a_2D_Ordinary_Differential_Equation.jl
#! /usr/bin/env julia

using DifferentialEquations

b, c = 0.25, 5.0
pend2 = @ode_def Pendulum begin
  dθ = ω
  dω = (-b * ω) - (c * sin(θ))
end

prob = ODEProblem(pend, y0, (0.0, 10.0))
sol = solve(prob)   # solve on [0,10]
t, y = sol.t, hcat(sol.u...)'

using Winston
plot(t, y[:, 1], t, y[:, 2])
title("2D Differential Equation")
savefig("figures/Pendulum_solution.png")

File: 2018_01__Python_introduction_for_MATLAB_users__at_Supelec/examples/Solving_a_2D_Ordinary_Differential_Equation.py
#! /usr/bin/env python
# -*- coding: utf8 -*-

import numpy as np
import matplotlib.pyplot as plt
from scipy.integrate import odeint  # use Runge-Kutta 4


def pend(y, t, b, c):  # function definition
    """Gives 2D vector dy/dt as function of y and t, with parameters b and c."""
    return np.array([y[1], -b*y[1] - c*np.sin(y[0])])


b, c = 0.25, 5.0  # tuple assignment
y0 = np.array([np.pi - 0.1, 0.0])
t = np.linspace(0, 10, 101)  # on [0,10] with 101 points

sol = odeint(pend, y0, t, args=(b, c))

plt.plot(t, sol[:, 0], 'b', label=r'$\theta(t)$')  # blue
plt.plot(t, sol[:, 1], 'g', label=r'$\omega(t)$')  # green
plt.legend(loc='best')
plt.xlabel('t')
plt.grid()
plt.savefig("figures/Pendulum_solution.png")
plt.show()
